services/libs/aster-std/src/sched/scheduler/rt_round_robin/mod.rs

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::prelude::*;
+use aster_frame::task::{NeedResched, ReadPriority, Scheduler, Task, TaskAdapter};
+use intrusive_collections::{linked_list::CursorMut, LinkedList};
+
+/// The preempt scheduler
+///
+/// Real-time tasks are placed in the `real_time_tasks` queue and
+/// are always prioritized during scheduling.
+/// Normal tasks are placed in the `normal_tasks` queue and are only
+/// scheduled for execution when there are no real-time tasks.
+pub struct PreemptiveRRScheduler {
+    /// Tasks with a priority of less than 100 are regarded as real-time tasks.
+    real_time_tasks: SpinLock<LinkedList<TaskAdapter>>,
+    /// Tasks with a priority greater than or equal to 100 are regarded as normal tasks.
+    normal_tasks: SpinLock<LinkedList<TaskAdapter>>,
+}
+
+impl PreemptiveRRScheduler {
+    pub fn new() -> Self {
+        Self {
+            real_time_tasks: SpinLock::new(LinkedList::new(TaskAdapter::new())),
+            normal_tasks: SpinLock::new(LinkedList::new(TaskAdapter::new())),
+        }
+    }
+
+    /// Remove a task from the queue
+    /// Returns true if the task is found and removed, false otherwise
+    fn rm_task_from_queue(task: &Arc<Task>, mut cursor: CursorMut<'_, TaskAdapter>) -> bool {
+        while let Some(t) = cursor.get() {
+            if t == task.as_ref() {
+                return cursor.remove().is_some();
+            }
+            cursor.move_next();
+        }
```

Comment:
The  `rm_task_from_queue` method seems  to be inefficient now. It will iterate over the `LinkedList` to find the exact task to remove. Ideally, this should be a `O(1)` operation instead of a `O(n)` operation.

The `LinkedList` provides a method [cursor_from_mut_ptr](https://docs.rs/intrusive-collections/latest/intrusive_collections/linked_list/struct.LinkedList.html#method.cursor_mut_from_ptr) for creating a cursor with `O(1)` complexity.

The code looks like 
```rust
let cursor_mut = unsafe { LinkedList::cursor_mut_from_ptr(&task as *const _); }
cursor_mut.remove();
```
But this is an unsafe operation. We may need to add some abstraction in `aster-frame`, for example, add a `remove_from_task_queue` method for `Task`.